AutoCollection/CorrelationContextManager.ts

```diff
@@ -168,7 +168,25 @@ export class CorrelationContextManager {
                 return AppInsightsAsyncCorrelatedErrorWrapper.apply(Object.create(AppInsightsAsyncCorrelatedErrorWrapper.prototype), arguments);
             }
 
+            // Is this object set to rewrite the stack?
+            // If so, we should turn off some Zone stuff that is prone to break
+            var stackRewrite = orig['stackRewrite'];
+            if (orig['prepareStackTrace']) {
+                orig['stackRewrite'] = false;
+                var stackTrace = orig['prepareStackTrace'];
+                orig['prepareStackTrace'] = (e, s) => {
+                    // Remove some AI and Zone methods from the stack trace
+                    // Otherwise we leave side-effects
+                    s.splice(0, 3);
```

Comment:
I don't expect the first 3 methods to change without a code change on our side, since they're all basically passing around our error handlers and zone error handlers. (In other words, the whole top of the stack is just AI SDK and Zone) If we ever change how our error handling works we'll have to adjust this number.